cnvlib/reports.py

```diff
@@ -86,9 +86,9 @@ def gainloss_by_segment(probes, segments, threshold, skip_low=False):
     """
     for segment, subprobes in probes.by_ranges(segments):
         if abs(segment.log2) >= threshold:
-            for (gene, chrom, start, end, _coverage, nprobes
-                ) in group_by_genes(subprobes, skip_low):
-                yield (gene, chrom, start, end, segment.log2, nprobes)
+            for (gene, chrom, start, end, _coverage, nprobes) in \
+                    group_by_genes(subprobes, skip_low):
+                yield segment._replace(gene=gene, start=start, end=end)
```

Comment:
This is causing the unit test to fail. The ordering of columns in the `gainloss` output is different, so keeping the extra columns is going to need a different strategy. (It also needs to be done the same with and without the `segments` input.) E.g.:

     columns = ['Gene', 'Chrom.', 'Start', 'End', 'Log2Ratio', 'Probes']
     extra_column_names = cnarr.data.columns.tolist()[len(cnarr._required_columns):]
     columns.extend(extra_column_names)